addons/phone_validation/models/phone_blacklist.py

```diff
@@ -40,15 +45,19 @@ def create(self, values):
             done.add(sanitized)
             to_create.append(dict(value, number=sanitized))
 
-        """ To avoid crash during import due to unique email, return the existing records if any """
-        sql = '''SELECT number, id FROM phone_blacklist WHERE number = ANY(%s)'''
-        numbers = [v['number'] for v in to_create]
-        self._cr.execute(sql, (numbers,))
-        bl_entries = dict(self._cr.fetchall())
-        to_create = [v for v in to_create if v['number'] not in bl_entries]
+        # Get existing phone blacklist (for requested numbers to create)
+        numbers_requested = [tc['number'] for tc in to_create]
+        pb_existing = self.search([('active', 'in', [True, False]), ('number', 'in', numbers_requested)])
```

Comment:
What is this "pb" ? Why not simply "existing" ? Also could use active_test key
```suggestion
        # search for existing entries, even inactive ones (will be activated again)
        existing = self.with_context(active_test=False).search([('number', 'in', numbers_requested)])
```